src/pymor/discretizations/iosys.py

```diff
@@ -0,0 +1,1702 @@
+            `estimator.estimate(U, mu, self)`.
+        visualizer
+            A visualizer for the problem. This can be any object with
+            a `visualize(U, discretization, ...)` method. If `visualizer`
+            is not `None`, a `visualize(U, *args, **kwargs)` method is added
+            to the discretization which forwards its arguments to the
+            visualizer's `visualize` method.
+        cache_region
+            `None` or name of the cache region to use. See
+            :mod:`pymor.core.cache`.
+        name
+            Name of the system.
+
+        Returns
+        -------
+        lti
+            The |LTISystem| with operators A, B, C, D, and E.
+        """
+        from pymor.tools.io import load_matrix
+        import os.path
+
+        A = load_matrix(files_basename + '.A')
+        B = load_matrix(files_basename + '.B')
+        C = load_matrix(files_basename + '.C')
+        D = load_matrix(files_basename + '.D') if os.path.isfile(files_basename + '.D') else None
+        E = load_matrix(files_basename + '.E') if os.path.isfile(files_basename + '.E') else None
+
+        return cls.from_matrices(A, B, C, D, E, cont_time=cont_time,
+                                 solver_options=solver_options, estimator=estimator, visualizer=visualizer,
+                                 cache_region=cache_region, name=name)
+
+    def __mul__(self, other):
+        """Multiply (cascade) two |LTISystems|."""
+        assert self.B.source == other.C.range
+
+        A = BlockOperator([[self.A, Concatenation(self.B, other.C)],
+                           [None, other.A]])
+        B = BlockOperator.vstack((Concatenation(self.B, other.D),
+                                  other.B))
+        C = BlockOperator.hstack((self.C,
+                                  Concatenation(self.D, other.C)))
+        D = Concatenation(self.D, other.D)
+        E = BlockDiagonalOperator((self.E, other.E))
+
+        return self.__class__(A, B, C, D, E, cont_time=self.cont_time)
+
+    @cached
+    def poles(self, force_dense=False):
+        """Compute system poles.
+
+        Parameters
+        ----------
+        force_dense
+            Should dense matrix computations be forced.
+        """
+        if not force_dense:
+            if not (isinstance(self.A, NumpyMatrixOperator) and not self.A.sparse):
+                raise TypeError('Expected A to be NumpyMatrixOperator with dense matrix (force_dense == False).')
+            if not ((isinstance(self.E, NumpyMatrixOperator) and not self.E.sparse) or
+                    _is_like_identity_operator(self.E)):
+                raise TypeError('Expected E to be NumpyMatrixOperator with dense matrix or IdentityOperator'
+                                '(force_dense == False).')
+
+        A = to_matrix(self.A, format='dense')
+        E = None if _is_like_identity_operator(self.E) else to_matrix(self.E, format='dense')
+        return spla.eigvals(A, E)
```

Comment:
Why can't the same be done for sparse problems?

Reply:
This would convert it into a dense problem, which would use a lot of memory and time. An option could be calling `scipy.sparse.linalg.eigs`.

Reply:
Yeah, it would make sense, but I'm not sure which would be good default arguments for `eigs` and how would `poles` handle the dense and sparse case in a nice way.
I've opened an issue about this in #389. Maybe we can postpone this discussion after the merge?

Reply:
Fine with me.